purchases/src/main/kotlin/com/revenuecat/purchases/common/offerings/OfferingsManager.kt

```diff
@@ -94,6 +95,7 @@ internal class OfferingsManager(
             },
             onSuccess = { offerings ->
                 offeringsCache.cacheOfferings(offerings, offeringsJSON)
+                offeringsCache.setOfferingsCacheTimestampToNow()
```

Comment:
This is another optimization I had to do... If we left it where it was before, we wouldn't use the new mechanisms to optimize the requests trying to use the request that is supposed to finish earlier.

Reply:
I wonder if these two lines should just be a part of cacheOfferings in offeringsCache and guarantee atomicity there

Reply:
Feels like it'd be conceptually cleaner and slightly easier to make it synchronized?

Reply:
Certainly, these 2 could be joined (this is the only place they are both called) I will do so in a different PR.